src/pages/en/guides/debugging.md

```diff
@@ -1,16 +1,118 @@
+
+### Unable to render Component
+
+This is indicates an error in a component you have imported and are using in your Astro template. 
+
+**Solution**: Check the appropriate documentation for your [Astro](/en/core-concepts/astro-components/) or [UI framework](/en/core-concepts/framework-components/) component. Consider opening an Astro starter template from [astro.new](https://astro.new) and troubleshooting just your component in a minimal Astro project. 
+
+> This can also be caused by attempting to access the `window` or `document` object at render time. 
+ 
+By default, Astro will render your component [isomorphically](https://en.wikipedia.org/wiki/Isomorphic_JavaScript), meaning it runs on the server where the browser runtime is not available. You can disable this pre-render step using [the `client:only` directive](/en/reference/directives-reference/#clientonly).
+
+**Solution**: Try to access those objects after rendering (ex: [`useEffect()`](https://reactjs.org/docs/hooks-reference.html#useeffect) in React or [`onMounted()`](https://vuejs.org/api/composition-api-lifecycle.html#onmounted) in Vue and Svelte)
+
+**Status**: Expected Astro behavior, as intended.
+
+💡 Not sure that this is your problem? Check to see if anyone else has reported [this issue](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Unable+to+render+Component)!
+
+
+### Expected a default export
+
+This error can be thrown when trying to import or render an invalid component, or one that is not working properly. (This particular message occurs because of the way importing a UI component works in Astro.)
+
+**Solution**: Try looking for errors in any component you are importing and rendering, and make sure it's working correctly. Consider opening an Astro starter template from [astro.new](https://astro.new) and troubleshooting just your component in a minimal Astro project. 
+
+**Status**: Expected Astro behavior, as intended.
+
+## Common "Gotchas" in Astro
+
+### My component is not rendering
+
+First, check to see that you have **imported the component** in your [`.astro` component script](/en/core-concepts/astro-components/#the-component-script) or [`.md` frontmatter](/en/guides/markdown-content/#using-components-in-markdown).
+
+Then check your import statement:
+
+- Is your import linking to the wrong place? (Check your import path.)
+
+- Does your import have the same name as the imported component? (Check your component name and that it [follows the `.astro` syntax](/en/comparing-astro-vs-other-tools/#astro-vs-jsx).)
+
+- Have you included the extension in the import? (Check that your imported file contains an extension. e.g. `.astro`, `.md`, `.jsx`, `.vue`)
+
+### My component renders on the page, but is not responding to user interaction
+
+If your component is rendering (see above) but is not interactive, then you may be missing a [`client:*` directive](/en/reference/directives-reference/#client-directives) to hydrate your component.
+
```

Comment:
```suggestion

By default, a [UI Framework component is not hydrated in the client](/en/core-concepts/framework-components//#hydrating-interactive-components). If no client:* directive is provided, its HTML is rendered onto the page without JavaScript.

Note: [Astro components](/en/core-concepts/astro-components/) are HTML-only templating components with no client-side runtime. But, you can use a <script> tag in your Astro component template to send JavaScript to the browser that executes in the global scope.

```

Reply:
This is copied from other sections, so it is duplicate content. BUT, it's maybe important enough to also have here because it's one of the MAIN THINGS people need to know? 

I feel like people used to Astro would see the one-liner, and "Duh, of course." BUT to make it friendlier to newer Astronauts, I think this might be a kind service to have these extra sound bites here.